thirdweb/core/classes/ipfs_storage.py

```diff
@@ -44,16 +44,22 @@ class IpfsStorage(ABC):
     def __init__(self, gateway_url=DEFAULT_IPFS_GATEWAY):
         self._gateway_url = re.sub(r"\/$", "", gateway_url) + "/"
 
-    def get(self, hash: str) -> Dict[str, Any]:
+    def get(self, hash: str) -> Any:
         """
-        Gets IPFS data at a given hash and returns it as a dictionary.
+        Gets IPFS data at a given hash and returns the data.
 
         :param hash: hash of the data to get.
-        :returns: dictionary of the data.
+        :returns: dictionary of the data if JSON, otherwise raw data.
         """
 
         res = self._get(hash)
-        data = res.json()
+
+        # If the data is not JSON, return it raw
+        try:
+            data = res.json()
+        except Exception:
+            return data
```

Comment:
This seems not related to custom contract? What was the motivation here. Also we need to be careful because in most languages, you can only consume the response once. So just trying to get the json out of it makes the response unusable after that

Reply:
Hmm probably better to leave it as it was before then for now - we can handle this properly in another PR, since we shouldn't just assume that the data is JSON (same goes for TS/Go SDK)